### DISCUSSION TITLE
Free plan current usage

### DISCUSSION THREAD

**Discussion by perissf1** [community/community#147945] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I am happily using Copilot GPT 4o free plan in VS Code!
Is it possible to see the current monthly usage?

**Comment by Akash1134** [comment, marked as answer]
Hi @perissf1 ,

With the new Copilot Free tier, the Code completions you get are limited to 2000 completions per month. Copilot Chat is limited to 50 chat messages per month. This limit includes both standard chats and multi-file editing chats in VS Code and Visual Studio. When you reach these limits, you can upgrade to Copilot Pro to continue using Copilot.

Regarding checking the usage, currently there is no way to actively track the usage. 

Hope that helps.

**Reply by waynehelleysewtec** [reply]
I feel like i must have sent well over 50 messages this month but it still carries on as normal for me.  Has anybody received a notification when hitting 50 messages?

**Reply by Enissay38** [reply]
hop it's unlimitted waiting someone saying " yea i reach the limite " but i prefere to not see it

**Reply by deshpandemanish-tech** [reply]
Hi @Akash1134  ,

I am getting "You've reached your monthly chat messages quota. Upgrade to Copilot Pro (30-day free trial) or wait for your allowance to renew." I followed the steps to upgrade my plan right from from providing billing information, payment details and finally click of activate plan. Even after that I am still getting the same error message. Any Idea why am i not able to upgrade from free to paid plan and if so how can it be resolved?
